File: src/main/java/com/back/domain/member/reputation/repository/ReportRepository.java
package com.back.domain.member.reputation.repository;

import com.back.domain.member.member.entity.Member;
import com.back.domain.member.reputation.entity.Report;
import org.springframework.data.jpa.repository.JpaRepository;

import java.time.LocalDateTime;

public interface ReportRepository extends JpaRepository<Report, Integer> {
    boolean existsByReporterAndTargetAndCreateDateAfter(
            Member reporter,
            Member target,
            LocalDateTime now);

    int countByReporterIdAndCreateDateAfter(int targetId, LocalDateTime now);
}

File: src/main/java/com/back/domain/member/reputation/repository/ReportRepository.kt
package com.back.domain.member.reputation.repository

import com.back.domain.member.member.entity.Member
import com.back.domain.member.reputation.entity.Report
import org.springframework.data.jpa.repository.JpaRepository
import java.time.LocalDateTime

interface ReportRepository : JpaRepository<Report, Int> {
    fun existsByReporterAndTargetAndCreateDateAfter(
        reporter: Member,
        target: Member,
        now: LocalDateTime
    ): Boolean

    fun countByReporterIdAndCreateDateAfter(targetId: Int, now: LocalDateTime): Int
}
